File: backend/src/main/java/com/fitapp/catalog/domain/TimerSettings.java
package com.fitapp.catalog.domain;

/**
 * `05-modelo-dominio-reglas.md` §1 (class TimerSettings, RN-05). Mirrors `TimerSettings` in
 * openapi.yaml.
 */
public record TimerSettings(
    Integer prepSeconds,
    Integer workSeconds,
    Integer restBetweenSetsSeconds,
    Integer restBetweenExercisesSeconds,
    Integer restBetweenRoundsSeconds,
    Boolean halfwayCue) {}

File: apps/mobile/src/shared/domain/TimerSettings.ts
/**
 * Forma estructural compartida (sin identidad propia), F02-T03
 * (`specs/F02-catalogo-propuesta/plan.md` §2 "Decisión de diseño"):
 * `05-modelo-dominio-reglas.md` §1 (class TimerSettings), RN-05.
 */
export interface TimerSettings {
  prepSeconds: number;
  workSeconds: number;
  restBetweenSetsSeconds: number;
  restBetweenExercisesSeconds: number;
  restBetweenRoundsSeconds: number;
  halfwayCue: boolean;
}
